addons/fleet/models/fleet_vehicle.py

```diff
@@ -0,0 +1,549 @@
+    doors = fields.Integer(string='Doors Number', help='Number of doors of the vehicle', default=5)
+    tag_ids = fields.Many2many('fleet.vehicle.tag', 'fleet_vehicle_vehicle_tag_rel', 'vehicle_tag_id', 'tag_id',
+                               string='Tags', copy=False)
+    odometer = fields.Float(compute='_compute_get_odometer', inverse='_compute_odometer', string='Last Odometer',
+                            help='Odometer measure of the vehicle at the moment of this log')
+    odometer_unit = fields.Selection([('kilometers', 'Kilometers'), ('miles', 'Miles')],
+                                     help='Unit of the odometer',
+                                     required=True, default='kilometers')
+    transmission = fields.Selection([('manual', 'Manual'), ('automatic', 'Automatic')],
+                                    help='Transmission Used by the vehicle')
+    fuel_type = fields.Selection([('gasoline', 'Gasoline'), ('diesel', 'Diesel'),
+                                            ('electric', 'Electric'), ('hybrid', 'Hybrid')],
+                                 help='Fuel Used by the vehicle')
+    horsepower = fields.Integer()
+    horsepower_tax = fields.Float(string='Horsepower Taxation')
+    power = fields.Integer(help='Power in kW of the vehicle')
+    co2 = fields.Float(string='CO2 Emissions', help='CO2 emissions of the vehicle')
+    image = fields.Binary(related='model_id.make_id.image', string="Logo", store=True)
+    image_medium = fields.Binary(related='model_id.make_id.image_medium', string="Logo (medium)", store=True)
+    image_small = fields.Binary(related='model_id.make_id.image_small', string="Logo (small)", store=True)
+    contract_renewal_due_soon = fields.Boolean(compute='_compute_get_contract_reminder',
+                                               search='_search_contract_renewal_due_soon',
+                                               string='Has Contracts to renew')
+    contract_renewal_overdue = fields.Boolean(compute='_compute_get_contract_reminder',
+                                              search='_search_get_overdue_contract_reminder',
+                                              string='Has Contracts Overdue')
+    contract_renewal_name = fields.Text(compute='_compute_get_contract_reminder', string='Name of contract to renew soon')
+    contract_renewal_total = fields.Integer(compute='_compute_get_contract_reminder',
+                                            string='Total of contracts due or overdue minus one')
+    car_value = fields.Float(help='Value of the bought vehicle')
+
+    _sql_constraints = [('unique_chassis_number', 'unique(vin_sn)', 'Same Chassis Number is already exists.')]
+
+    @api.one
+    @api.depends('license_plate')
+    def _compute_vehicle_name(self):
+        self.name = ('%s' + ' / ' + '%s' + ' / ' + '%s') % (self.model_id.make_id.name, self.model_id.name, self.license_plate)
+
+    @api.multi
+    def _compute_get_contract_reminder(self):
```

Comment:
Follow the name pattern at all places in this branch for,

compute functions = _compute_<field_name>
inverse functions = _inverse_<field_name>
default functions = _default_<field_name>
onchange function = _onchange_<field_name>